### DISCUSSION TITLE
Is it possible to disable load at depots?

### DISCUSSION THREAD

**Discussion by pocket7878** [reinterpretcat/vrp#83] [Q&A] [answered]
We try to solve a problem with multiple-pick-up & multiple deliveries and no goods at depot.
We want to find a tour like a 「start from depot, picking-up 2 goods, drop-off 1 of them, next picking-up 1 goods, picking up 2 goods, drop-off 2 of them...」
We are allowed to tour ended-up with goods left on the vehicle. (we are collecting left goods at depot)

First we tried to solve as separate each pick-up & delivery as independent jobs, but then we got a solution with load at start depot. But we didn't have any goods at depot, we should picking-up it as a job.
Next we tried to solve with gathering all pick-up & deliveries into a single job, but then we are not able to allow to tour ended-up with goods left on the vehicle.

**Comment by reinterpretcat** [comment, marked as answer]
Start has no load in the following cases:
- only pickup jobs are used
- only pickup & delivery jobs are used (known as multi-jobs, see as example https://github.com/reinterpretcat/vrp/blob/master/examples/data/pragmatic/basics/multi-job.basic.problem.json#L4-L55)
- mix of two cases above

However, only single pickup job will have some goods kept to the tour end and p&d jobs have assignment constraints (e.g. all or none of them are assigned)

It seems, what you need is p&d multi-job with optional deliveries. This is not supported. What you can try as workaround:

- use only p&d multi jobs
- use tour order feature to lock some specific deliveries to the end of the tour:

    > tour-order: controls desired activity order in tours
        isConstrained: violating order is not allowed, even if it leads to less assigned jobs (default is true).

    NOTE: I need to update the documentation to reflect the change in default value (isConstrained=true)

    https://reinterpretcat.github.io/vrp/concepts/pragmatic/problem/objectives.html#job-distribution-objectives
- if deliveries cannot be assigned due to some constraint (e.g. time), relax these constraints (e.g. shift time)
- do post processing and remove delivery stops, which violate constraints, from the end of the tour

**Reply by pocket7878** [reply]
Thank you for your quick and polite response! We'll try to suggested workarounds.
